### ISSUE TITLE
mako stopped showing notifications after powering display off and on

### ISSUE THREAD

**Issue by kchibisov** [YaLTeR/niri#23]
it sort of fixed itself with reloading mako. I'm not sure who is at fault here, but unlikely mako given that it worked on sway just fine.

**Comment by kchibisov** [comment]
I think the issue is basically with enter/leave the output, since `tty` switch hides mako notifications. But maybe it's something expected? Haven't tested with sway yet.

**Comment by YaLTeR** [comment]
I've seen it once myself a few days ago but since then it worked fine for me, and I do plenty of tty switching for testing. Maybe something timing-related?

**Comment by kchibisov** [comment]
Hm, maybe. But for me it's reliable... The thing is that you should have an open notification iirc.

**Comment by YaLTeR** [comment]
Oh, hm, I should try that.

**Comment by YaLTeR** [comment]
Yeah, if you replug the monitor while a notification is showing on that monitor, it breaks. I'm not sure whose fault this is.

**Comment by YaLTeR** [comment]
Seems that sway sends layer-surface.closed() in this case.

**Comment by YaLTeR** [comment]
Okay, yeah, that was the problem.

**Comment by kchibisov** [comment]
I mean yeah, if output is removed you should remove the surfaces.
